flink-runtime/src/main/java/org/apache/flink/runtime/entrypoint/ClusterConfiguration.java

```diff
@@ -18,27 +18,43 @@
 
 package org.apache.flink.runtime.entrypoint;
 
-import org.apache.flink.util.Preconditions;
+import javax.annotation.Nonnull;
+
+import java.util.Properties;
 
 /**
  * Configuration class which contains the parsed command line arguments for
  * the {@link ClusterEntrypoint}.
  */
 public class ClusterConfiguration {
+
+	@Nonnull
 	private final String configDir;
 
-	private final int restPort;
+	@Nonnull
+	private final Properties dynamicProperties;
+
+	@Nonnull
+	private final String[] args;
 
-	public ClusterConfiguration(String configDir, int restPort) {
-		this.configDir = Preconditions.checkNotNull(configDir);
-		this.restPort = restPort;
+	public ClusterConfiguration(@Nonnull String configDir, @Nonnull Properties dynamicProperties, @Nonnull String[] args) {
```

Comment:
In `FlinkParseException` it does not follow the convention because the parent class also does not follow the convention. In the `ClusterConfiguration` the usage of `@Nonnull` should be consistent and allows us to not explicitly have to check for non null arguments.